src/tribler-core/tribler_core/components/tag/rules/tag_rules_processor.py

```diff
@@ -0,0 +1,109 @@
+import logging
+from typing import Optional, Set
+
+from ipv8.taskmanager import TaskManager
+
+from pony.orm import db_session
+
+import tribler_core.components.metadata_store.db.orm_bindings.torrent_metadata as torrent_metadata
+import tribler_core.components.metadata_store.db.store as MDS
+from tribler_core.components.metadata_store.db.serialization import REGULAR_TORRENT
+from tribler_core.components.tag.community.tag_payload import TagOperation, TagOperationEnum
+from tribler_core.components.tag.db.tag_db import (
+    CLOCK_START_VALUE, PUBLIC_KEY_FOR_AUTO_GENERATED_TAGS,
+    TagDatabase,
+)
+from tribler_core.components.tag.rules.tag_rules import extract_only_valid_tags
+from tribler_core.notifier import Notifier
+
+LAST_PROCESSED_TORRENT_ID = 'last_processed_torrent_id'
+
+
+class TagRulesProcessor(TaskManager):
```

Comment:
**Suggestion:** It seems that we now apply rules when 1) tags are coming in from the network, and 2) passively in the background. We might consider also extracting tags for unprocessed torrents when the user requests torrents with the REST API (if it's not adding too much latency). This would ensure that there are tags visible for the content that users see. The next time the user requests the same torrents, the rules are already applied to the requested torrents.

Reply:
My previous version of this PR had this logic.

I abandoned this logic because assumed that (1) and (2) are good enough as (1) works for all new torrents and (2) processes the whole DB within the first day after the update.

So I decided not to add a third point of roughly the same logic that will be useful only within the first hours after the update.

But if @devos50 and @kozlovsky vote to return the logic back, I'm ok with it.

Reply:
I think it might be good to add tags when retrieving torrents via REST API, as it will demonstrate tags to the user immediately after he starts using the new version of Tribler.

Reply:
As an alternative, we might show the user that tags are being processed in the background so the user is aware that something is happening, and more tags will be visible after the processor is finished (we did the same when updating from legacy channels). Still, I prefer for tags to work without waiting for the completion of the processor and it would improve the user experience, even if it is logic that is only useful a few hours after an upgrade.

Reply:
Done